### PR TITLE
feature/se-68

### PR DESCRIPTION
**Pull request by Markus799**

### CODE REVIEW

File: api-models/job.yaml [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+          schema:
+            type: array
+            items:
+              type: string
+        - name: filterByJobName
+          in: query
+          description: String which should be part of the job name
+          required: false 
+          explode: true 
+          schema:
```
Comment (julix14) [inline api-models/job.yaml:29]: We agreed on don't offer further filter options

File: api-models/job.yaml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            type: string
+        - name: nested
```
Comment (julix14) [inline api-models/job.yaml:31]:
```suggestion
        - name: full
```

File: api-models/job.yaml [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  /jobs/{jobId}:
+    get:
+      summary: Get a job by ID
+      operationId: getJobByID
+      parameters:
+        - name: jobId
+          in: path
+          required: true
+          schema:
+            type: string
```
Comment (julix14) [inline api-models/job.yaml:70]: Add "full" option here as well

File: api-models/job.yaml [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Job'
+      responses:
+        '200':
+          description: Job updated
+        '404':
+          description: Job not found
```
Comment (julix14) [inline api-models/job.yaml:100]: Add response for "invalid request"
        '400':
          description: Invalid request data

File: api-models/job.yaml [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+paths:
+  /jobs:
+    get:
+      summary: List all jobs
+      operationId: getAllCourses
+      parameters: 
+        - name: only
+          in: query
+          description: Array of course IDs to fetch ta subset of jobs
```
Comment (julix14) [inline api-models/job.yaml:16]:
```suggestion
          description: Array of job IDs to fetch a subset of jobs
```

File: api-models/job.yaml [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        - name: filterByJobName
+          in: query
+          description: String which should be part of the job name
+          required: false 
+          explode: true 
+          schema:
+            type: string
+        - name: full
+          in: query
+          description: Determise if the nested data will be returnd
```
Comment (julix14) [inline api-models/job.yaml:33]:
```suggestion
          description: Determine if the nested data will be returned
```

File: api-models/job.yaml [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+            type: string
+      responses:
+        '200':
+          description: Job details
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Job'
+        '400':
+          description: Coruse not found
```
Comment (julix14) [inline api-models/job.yaml:79]:
```suggestion
          description: Job not found
```

File: api-models/job.yaml [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        skills:
+          type: array
+          items: 
+            type: string
+          description: Array of skills - See more in the skills microservice
+        tags:
+          type: array
+          items:
+            type: string
+          description: Array of tags - See mor in the skills microservice
```
Comment (julix14) [inline api-models/job.yaml:143]:
```suggestion
          description: Array of tags - See more in the skills microservice
```

File: api-models/job.yaml [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+  version: "1.0.0"
+  title: Job Microservice API
+  description: >-
+    This API manages different types of jobs in SkillVerse
+
+paths:
+  /jobs:
+    get:
+      summary: List all jobs
+      operationId: getAllCourses
```
Comment (julix14) [inline api-models/job.yaml:12]:
```suggestion
      operationId: getAllJobs
```

File: api-models/job.yaml (hunk 8 above)
Comment (julix14) [inline api-models/job.yaml:143]:
```suggestion
          description: Array of tags - See mor in the tags microservice
```

File: api-models/job.yaml [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+      operationId: getJobByID
+      parameters:
+        - name: jobId
+          in: path
+          required: true
+          schema:
+            type: string
+        - name: full
+          in: query
+          description: Determise if the nested data will be returnd
```
Comment (julix14) [inline api-models/job.yaml:66]:
```suggestion
          description: Determine if the nested data will be returned
```

File: api-models/job.yaml [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+            schema:
+              $ref: '#/components/schemas/Job'
+      responses:
+        '201':
+          description: Job created
+  
+  /jobs/{jobId}:
+    get:
+      summary: Get a job by ID
+      operationId: getJobByID
```
Comment (julix14) [inline api-models/job.yaml:57]:
```suggestion
      operationId: getJobById
```

**Review by julix14 — APPROVED** [review]

**Review by Chrissydrx — APPROVED** [review]

**Pull request merged**